wasi-filesystem.wit.md

```diff
@@ -816,6 +797,54 @@ try-lock-exclusive: func() -> result<_, errno>
 unlock: func() -> result<_, errno>
 ```
 
+## `is-same-object`
+```wit
+/// Test whether two descriptors refer to the same filesystem object.
+///
+/// In POSIX, this corresponds to testing whether the two descriptors have the
+/// same device (`st_dev`) and inode (`st_ino` or `d_ino`) numbers.
+/// wasi-filesystem does not expose device and inode numbers, so this function
+/// may be used instead.
+is-same-object: func(other: descriptor) -> bool
+```
+
+## `metadata-hash`
+```wit
+/// Return a hash of the metadata associated with a filesystem object referred
+/// to by a descriptor.
+///
+/// This returns a hash of the last-modification timestamp and file size, and
+/// may also include the inode number, device number, birth timestamp, and
+/// other metadata fields that may change when the file is modified or
+/// replaced.
+///
+/// Implementations are encourated to provide the following properties:
+///
+///  - If the file is not modified or replaced, the computed hash value should
+///    usually not change.
+///  - If the object is modified or replaced, the computed hash value should
+///    usually change.
+///  - The inputs to the hash should not be easily computable from the
+///    computed hash.
+///
+/// However, none of these is required.
```

Comment:
does this mean it can return a constant?

Reply:
Yes. I don't know of a way to rule that out without being excessively restrictive. But we can consider implementations that do that to have poor Quality Of Implementation. The current text does say "If the object is modified or replaced, the computed hash value should usually change."